addons/stock/models/stock_picking.py

```diff
@@ -771,6 +771,8 @@ def action_assign(self):
     def action_cancel(self):
         self.mapped('move_lines')._action_cancel()
         self.write({'is_locked': True})
+        if not self.mapped('move_lines'):
```

Comment:
You need filter to set the state only on picking with no moves.
`self.filtered(lambda x: not x.move_lines).state = 'cancel'`

Reply:
thanks and updated @fmdl